scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/JobVariable.java

```diff
@@ -91,56 +105,70 @@ public void setModel(String model) {
         this.model = model;
     }
 
-    @Override
-    public boolean equals(Object object) {
-        if (this == object) {
-            return true;
-        }
-        if (object == null) {
+    public String getDescription() {
+        return description;
+    }
+
+    public void setDescription(String description) {
+        this.description = description;
+    }
+
+    public String getGroup() {
+        return group;
+    }
+
+    public void setGroup(String group) {
+        this.group = group;
+    }
+
+    public boolean getAdvanced() {
+        if (advanced == null) {
             return false;
+        } else {
+            return advanced;
         }
-        if (getClass() != object.getClass()) {
+    }
+
+    public void setAdvanced(Boolean advanced) {
+        this.advanced = advanced;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o)
+            return true;
+        if (o == null || getClass() != o.getClass())
             return false;
-        }
 
-        JobVariable jobVariable = (JobVariable) object;
-        if (name == null) {
-            if (jobVariable.name != null) {
-                return false;
-            }
-        } else if (!name.equals(jobVariable.name)) {
+        JobVariable that = (JobVariable) o;
+
+        if (getName() != null ? !getName().equals(that.getName()) : that.getName() != null)
             return false;
-        }
-        if (value == null) {
-            if (jobVariable.value != null) {
-                return false;
-            }
-        } else if (!value.equals(jobVariable.value)) {
+        if (getValue() != null ? !getValue().equals(that.getValue()) : that.getValue() != null)
             return false;
-        }
-        if (model == null) {
-            if (jobVariable.model != null) {
-                return false;
-            }
-        } else if (!model.equals(jobVariable.model)) {
+        if (getModel() != null ? !getModel().equals(that.getModel()) : that.getModel() != null)
             return false;
-        }
-        return true;
+        if (getDescription() != null ? !getDescription().equals(that.getDescription()) : that.getDescription() != null)
+            return false;
+        if (getGroup() != null ? !getGroup().equals(that.getGroup()) : that.getGroup() != null)
+            return false;
+        return getAdvanced() == that.getAdvanced();
     }
 
     @Override
     public int hashCode() {
-        final int primeNumber = 31;
-        int result = 1;
-        result = primeNumber * result + ((name == null) ? 0 : name.hashCode());
-        result = primeNumber * result + ((value == null) ? 0 : value.hashCode());
-        result = primeNumber * result + ((model == null) ? 0 : model.hashCode());
+        int result = getName() != null ? getName().hashCode() : 0;
```

Comment:
Need a brief explanation about the hash function

Reply:
this is an auto-generated hashcode method